backend/game/engine/board.py

```diff
@@ -57,12 +54,12 @@ def re_roll(self, indexes_to_re_roll):
 
     def post_roll_actions(self, active_player):
         # TODO, refactor to other method
-        if active_player.has_instance_of_card(EnergyHoarder()):
-            EnergyHoarder.special_effect(
-                active_player, self.players.get_all_alive_players_minus_current_player())
```

Comment:
I think if you either set the card as an instance and call special effect, it should work as intended

`EnergyHoarder().special_effect(active_player, self.players.get_all_alive_players_minus_current_player())`

Reply:
Initialized as suggested. This PR now fixes #234